test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -2090,7 +2090,7 @@ def decode_value(cls, value):
         :type value: str or int
         :returns: int or float or string
         """
-        if re.search('\'{\".*\":.*', str(value)):
+        if re.search('(\{.*:.*[,]?.*\})', str(value)):
             value = eval(eval(str(value)))
```

Comment:
Why 2 evals ? It seems to fail for me:
>>> mystr = "{1:\"blah\", 2:[1, 'a', \"blah\"], 3: {1:2, 3:4}}"
>>> eval(str(mystr))
{1: 'blah', 2: [1, 'a', 'blah'], 3: {1: 2, 3: 4}}
>>> eval(eval(str(mystr)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: eval() arg 1 must be a string, bytes or code object
>>>

Reply:
With single eval statement we get this:
`2019-12-11 03:45:50,520 INFO     expect on server pbs-server@/etc/pbs.conf2: resources_used.foo_i = 29 && resources_used.foo_f = 0.29 && resources_used.foo_str = '{"eight": 8, "seven": 7, "nine": 9}' job 60.pbs-server attempt: 60  got: resources_used.foo_str = {"eight": 8, "nine": 9, "seven": 7}`

This is because with single eval type(value) is still string. That is why I have used double eval.

Reply:
I would suggest using `json.loads` instead eval, as "eval" is "evil".